revocation.md

```diff
@@ -0,0 +1,237 @@
+# Revocation
+
+**Patent Pending**
+
+Revocation requires that a verifiable revocation notice for a given Doughnut be posted somewhere
+accessible to all Doughnut revokers and verifiers.
+
+Because Doughnuts support decentralisation it makes sense for a Doughnut to tell verifiers
+how and where to revoke it, and where to check its revocation status.
+This removes the need for centralised revocation services.
+
+What follows are specifications for decentralised Doughnut revocation methods.
+A Doughnut version may specify a specific revocation method, or allow the issuer to
+make a choice of revocation method.
+
+Different strategies for achieving revocation are explored in the [`Strategies`](#strategies) section below.
+
+
+
+
+
+# Methods
+
+
+
+## 0
+Method 0 specifies a basic HTTP revocation interface with issuer/holder revocation.
+
+
+### Hashing
+The hashes specified below are BLAKE2 unless otherwise stated.
+
+
+### Doughnut
+The Doughnut will store a URI that points to a valid Method 0 Revocator.
+
+```
+    ...
+    revocation_uri: String
+    ...
+```
+
+
+### Revocator API
+
+A Method 0 Revocator is expected to provide the following HTTP(S) interface.
+
+#### `POST /revoke`
+
+An endpoint for revoking a Doughnut.
+
+##### Body
+```
+{
+    doughnut: String,
```

Comment:
Add to that, a revoker should be able to revoke all the doughnuts they have issued to a particular holder. Again, in this case, there would be no need to include a specific doughnut.